builder/azure/arm/step_create_resource_group.go

```diff
@@ -46,13 +48,17 @@ func (s *StepCreateResourceGroup) createResourceGroup(ctx context.Context, resou
 	return err
 }
 
-func (s *StepCreateResourceGroup) doesResourceGroupExist(ctx context.Context, resourceGroupName string) (bool, error) {
-	exists, err := s.client.GroupsClient.CheckExistence(ctx, resourceGroupName)
+func (s *StepCreateResourceGroup) doesResourceGroupExist(ctx context.Context, subscriptionId string, resourceGroupName string) (bool, error) {
+	id := commonids.NewResourceGroupID(subscriptionId, resourceGroupName)
+	exists, err := s.client.ResourceGroupsClient.Get(ctx, id)
 	if err != nil {
+		if exists.HttpResponse.StatusCode == 404 {
```

Comment:
Is there a chance that `HttpResponse` might be `nil` here? If so this will panic.

Reply:
Yes but this was true in the old SDK and it never happened, definitely worth guarding against but unlikely to happen I believe